### DISCUSSION TITLE
How can I get the Claims from an expired token

### DISCUSSION THREAD

**Discussion by oSethoum** [aidantwoods/go-paseto#36] [Q&A] [answered]
I have a use case where I refresh the access token using the refresh token, I need to check that both token have the same userId in the claims in order to allow the refresh, how can I do that?

**Comment by oSethoum** [comment, marked as answer]
found it 
```go
paseto.NewParserWithoutExpiryCheck()
```

**Comment by aidantwoods** [comment]
You don't have to use the "without expiry check" parser to check additional claims (only use that one if your token doesn't have an expiry claim). You can call `.AddRule` on the parser to add additional checks 🙂
